### PR TITLE
Fix cuEquivariance fusion device detection

### PR DESCRIPTION
**Pull request by zhubonan**
## Summary
- normalize the conversion device through `torch.device(device).type` before enabling convolution fusion
- preserve existing behavior for string CUDA devices while fixing `torch.device("cuda")`, which is passed by training
- add a focused regression test for both accepted device representations

## Why
The previous string comparison disabled cuEquivariance convolution fusion when the training code passed a `torch.device`. Local RTX 5090 profiling showed the corrected converted route matches native `--only_cueq` throughput within measurement noise.

## Testing
- `python -m pytest tests/unit/test_convert_e3nn_cueq.py tests/backends/test_cueq.py -q` (28 passed)
- `python -m py_compile mace/cli/convert_e3nn_cueq.py tests/unit/test_convert_e3nn_cueq.py`
- `git diff --check origin/develop...HEAD`

### CODE REVIEW

**Comment by zhubonan** [comment]
It seems that conv_fusion is detrimental for training as atoms/edges changes every batch. Perhaps it should be turned off by default instead?

**Comment by aacostadiaz** [comment]
@cursor review

**Review by aacostadiaz — APPROVED** [review]
